### PR TITLE
EVG-12820: Add pagination for system metrics rest routes

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-12820

This adds pagination for system metrics to the rest package.

### CODE REVIEW

File: rest/system_metrics_routes.go [hunk 1]
```diff
@@ -40,16 +43,23 @@ func (h *systemMetricsGetByTypeHandler) Factory() gimlet.RouteHandler {
 func (h *systemMetricsGetByTypeHandler) Parse(_ context.Context, r *http.Request) error {
 	var err error
 
-	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
-	h.metricType = gimlet.GetVars(r)["type"]
+	h.findOpts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.downloadOpts.MetricType = gimlet.GetVars(r)["type"]
 	vals := r.URL.Query()
+	h.downloadOpts.PageSize = softSizeLimit
```
Comment (Kimchelly) [inline rest/system_metrics_routes.go:49]: Should users be able to specify this in their request?

File: rest/data/system_metrics_test.go [hunk 2]
```diff
@@ -105,41 +106,65 @@ func (s *systemMetricsConnectorSuite) TearDownSuite() {
...
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
+	s.Require().NoError(err)
+	s.Equal("chunk2", string(data))
+	s.Equal(2, nextIdx)
+
+	downloadOpts.StartIndex = 2
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
+	s.Require().NoError(err)
+	s.Nil(data)
+	s.Equal(2, nextIdx)
```
Comment (Kimchelly) [inline rest/data/system_metrics_test.go:144]: Should this also test the case where the page size is large enough to read multiple chunks (but not all of them at once)?

File: rest/system_metrics_routes.go (hunk 1 above)
Comment (julianedwards) [reply rest/system_metrics_routes.go:49]: I don't think so, I don't want to trust users to set reasonable page sizes. We already set it for buildlogger and it seems to work fine.

File: rest/data/system_metrics_test.go (hunk 2 above)
Comment (julianedwards) [reply rest/data/system_metrics_test.go:144]: That is tested in the model package, I just want to test that it is calling the right thing.

**Pull request merged**